### PR TITLE
Render vimeo and youtube videos

### PR DESCRIPTION
**Pull request by TornikeW**
Render vimeo and youtube videos on notification feed

### CODE REVIEW

File: src/components/VideoPlayer/consts.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+export enum VideoSources {
+  YOUTUBE,
+  VIMEO,
+}
+
+export const videoSourcesInfo = {
+  [VideoSources.YOUTUBE]: {
+    baseUrl: 'https://www.youtube.com/embed/',
+    getIdRegex: /^.*((youtu.be\/)|(v\/)|(\/u\/\w\/)|(embed\/)|(watch\?))\??v?=?([^#&?]*).*/,
```
Comment (nivo33) [inline src/components/VideoPlayer/consts.ts:9]:
```suggestion
   idRegex: /^.*((youtu.be\/)|(v\/)|(\/u\/\w\/)|(embed\/)|(watch\?))\??v?=?([^#&?]*).*/,
```

File: src/components/VideoPlayer/index.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    url.match(videoSourcesInfo[VideoSources.VIMEO].getIdRegex)
+  );
+};
+
+export const getYoutubeId = (url: string) => {
+  const match = url.match(videoSourcesInfo[VideoSources.YOUTUBE].getIdRegex);
+  return match && match[7].length == 11 ? match[7] : '';
+};
+
+export const getVimeoId = (url: string) => {
```
Comment (nivo33) [inline src/components/VideoPlayer/index.tsx:41]: if these are also used in other places lets put them in a `helpers` file in this folder

File: src/components/VideoPlayer/index.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+  return (
+    url.match(videoSourcesInfo[VideoSources.YOUTUBE].getIdRegex) ||
+    url.match(videoSourcesInfo[VideoSources.VIMEO].getIdRegex)
+  );
+};
+
+export const getYoutubeId = (url: string) => {
+  const match = url.match(videoSourcesInfo[VideoSources.YOUTUBE].getIdRegex);
+  return match && match[7].length == 11 ? match[7] : '';
```
Comment (nivo33) [inline src/components/VideoPlayer/index.tsx:38]: you can use non-capturing groups to have it always be `match[1]`, and with the right regex you shouldn't need to check the length is 11, you can do that within the regex. Then you can also have a single `getVideoId` function :v:

https://stackoverflow.com/questions/3512471/what-is-a-non-capturing-group-in-regular-expressions

File: src/components/VideoPlayer/index.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+const VideoPlayer = ({ url }: { url: string }) => (
+  <Container>
+    <iframe
+      height="180px"
+      src={buildUrl(url)}
+      frameBorder="0"
+      allow="fullscreen; picture-in-picture"
+      allowFullScreen
+      title="Embedded video"
```
Comment (nivo33) [inline src/components/VideoPlayer/index.tsx:67]: just cos this is how we're doing it everywhere
```suggestion
      height={"180px"}
      src={buildUrl(url)}
      frameBorder={"0"}
      allow={"fullscreen; picture-in-picture"}
      allowFullScreen
      title={"Embedded video"}
```

File: src/components/VideoPlayer/index.tsx (hunk 2 above)
Comment (TornikeW) [reply src/components/VideoPlayer/index.tsx:41]: @nivo33 They are not forgot to remove export, will just remove it

File: .eslintrc.json [hunk 5]
```diff
@@ -44,7 +44,8 @@
     "react-hooks/rules-of-hooks": "error",
     "react-hooks/exhaustive-deps": "warn",
     "no-use-before-define": "warn",
-    "@typescript-eslint/no-var-requires": "off"
+    "@typescript-eslint/no-var-requires": "off",
+    "no-useless-escape": "off" // Required for regex warnings
```
Comment (nivo33) [inline .eslintrc.json:48]: dont think it should be required. if its giving you error it means there was a usesless `\` escape no? 🙃

Comment (nivo33) [reply .eslintrc.json:48]: Also this is a json file, json doesn't have comments so I think you need to remove the comment

**Review by nivo33 — APPROVED** [review]

**Pull request merged**